dap-mode.el

```diff
@@ -1236,6 +1242,8 @@ before starting the debug process."
 
 (defun dap--after-initialize ()
   "After initialize handler."
+  (when dap-auto-configure
```

Comment:
dap-auto-configure seems to be not needed and the dap-auto-configure-mode should be called after lsp-managed-mode has been enabled. So, generally, we might move calling auto-configuration mode in lsp-mode.